File: app/src/main/java/com/lazylee/lzywanandroid/data/dao/UserDao.java
package com.lazylee.lzywanandroid.data.dao;

import com.lazylee.lzywanandroid.data.entity.User;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

@Dao
public interface UserDao {

    @Query("Select * From User Where username Like :userName")
    User findByUserName(String userName);

    @Query("Select * From User Where username Like :userName And password Like :password")
    User findByUserNameAndPassword(String userName,String password);

    @Insert
    void insertUser(User user);

    @Delete
    void deleteUser(User user);
}

File: app/src/main/java/com/lazylee/lzywanandroid/data/dao/UserDao.kt
package com.lazylee.lzywanandroid.data.dao

import com.lazylee.lzywanandroid.data.entity.User

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface UserDao {

    @Query("Select * From User Where username Like :userName")
    fun findByUserName(userName: String): User

    @Query("Select * From User Where username Like :userName And password Like :password")
    fun findByUserNameAndPassword(userName: String, password: String): User

    @Insert
    fun insertUser(user: User)

    @Delete
    fun deleteUser(user: User)
}
